src/models/storage/batch.py

```diff
@@ -18,31 +18,37 @@
 from pandas import DataFrame
 
 from src.models.inference.outcome import Outcome
+from src.utils.logging_manager import LoggingManager, LoggingLevel
 
 
 class Batch:
     """
     Data model used for storing a batch of frames
 
     Arguments:
-        frames (DataFrame): List of video frames
-        info (FrameInfo): Information about the frames in the batch
+        frames (List[Dict], DataFrame): List of frames
         outcomes (Dict[str, List[BasePrediction]]): outcomes of running a udf
         with name 'x' as key
         identifier_column (str): A column used to uniquely a row
 
 
     """
 
-    def __init__(self, frames: DataFrame, outcomes=None, temp_outcomes=None,
+    def __init__(self, frames, outcomes=None, temp_outcomes=None,
                  identifier_column='id'):
         super().__init__()
         if outcomes is None:
             outcomes = dict()
         if temp_outcomes is None:
             temp_outcomes = dict()
-
-        self._frames = frames
+        if isinstance(frames, DataFrame):
+            self._frames = frames
+        elif isinstance(frames, list):
```

Comment:
I don't think this check is as simple as checking for type. 
For instance, ["string", 1, 2..] is a list. Also, [{'name': 1, 'data':[]}, {'col2': '..', 'col3': ...}] is a list.

I feel it's much easier to convey that the Batch API takes as input a pd.DataFrame. How it created is taken care by the API user outside.

Reply:
Done